plugins/inputs/irqstat/irqstat.go

```diff
@@ -0,0 +1,114 @@
+package irqstat
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type Irqstat struct{}
+
+type IRQ struct {
+	ID     string
+	Type   string
+	Device string
+	Values map[string]interface{}
+}
+
+func NewIRQ(id string) *IRQ {
+	return &IRQ{ID: id, Values: make(map[string]interface{})}
+}
+
+const sampleConfig = `
+  ## To filter which IRQs to collect, make use of tagpass / tagdrop, i.e.
+  # [inputs.irqstat.tagdrop]
+    # irq = [ "NET_RX", "TASKLET" ]
+`
+
+func (s *Irqstat) Description() string {
+	return "This plugin gathers interrupts data from /proc/interrupts and /proc/softirqs."
+}
+
+func (s *Irqstat) SampleConfig() string {
+	return sampleConfig
+}
+
+func parseInterrupts(irqdata string) ([]IRQ, error) {
+	var irqs []IRQ
+	var cpucount int
+	scanner := bufio.NewScanner(strings.NewReader(irqdata))
+	for scanner.Scan() {
+		var irqval, irqtotal int64
+		var irqtype, irqdevice string
+		fields := strings.Fields(scanner.Text())
+		if len(fields) > 0 && fields[0] == "CPU0" {
+			cpucount = len(fields)
+		}
+		if !strings.HasSuffix(fields[0], ":") {
+			continue
+		}
+		irqid := strings.TrimRight(fields[0], ":")
+		irq := NewIRQ(irqid)
+		_, err := strconv.ParseInt(irqid, 10, 64)
+		if err == nil && len(fields) > cpucount {
+			irqtype = fields[cpucount+1]
+			irqdevice = strings.Join(fields[cpucount+2:], " ")
+		} else if len(fields) > cpucount {
+			irqtype = strings.Join(fields[cpucount+1:], " ")
+		}
+		fields = fields[1:len(fields)]
+		for i := 0; i < cpucount; i++ {
+			cpu := fmt.Sprintf("CPU%d", i)
+			if i < len(fields) {
+				irqval, err = strconv.ParseInt(fields[i], 10, 64)
+				if err != nil {
+					return irqs, fmt.Errorf("Unable to parse %q from %q: %s", fields[i], scanner.Text(), err)
+				}
+				irq.Values[cpu] = irqval
+			}
+			irqtotal = irqval + irqtotal
+		}
+		irq.Type = irqtype
+		irq.Device = irqdevice
+		irq.Values["total"] = irqtotal
+		irqs = append(irqs, *irq)
+	}
+	return irqs, nil
+}
+
+func (s *Irqstat) Gather(acc telegraf.Accumulator) error {
+	files := []string{"/proc/interrupts", "/proc/softirqs"}
+	for _, file := range files {
+		data, err := ioutil.ReadFile(file)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Reading %s: %s", file, err))
+			continue
+		}
+		irqdata := string(data)
+		irqs, err := parseInterrupts(irqdata)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Parsing %s: %s", file, err))
+			continue
+		}
+		for _, irq := range irqs {
+			tags := map[string]string{"irq": irq.ID, "type": irq.Type, "device": irq.Device}
+			fields := irq.Values
+			if file == "/proc/softirqs" {
+				acc.AddFields("soft_interrupts", fields, tags)
+			} else {
+				acc.AddFields("interrupts", fields, tags)
+			}
+		}
+	}
+	return nil
+}
+
+func init() {
+	inputs.Add("irqstat", func() telegraf.Input {
```

Comment:
Sounds like consensus has been made, but since I was pinged directly: I think I would favor "interrupts". Name should be reflective of what it provides, not where it gets the data. Though one could argue that "irqstat" is short for "IRQ statistics", which is reflective of what it provides.